include/deal.II/lac/trilinos_tpetra_vector.templates.h

```diff
@@ -1032,12 +1032,14 @@ namespace LinearAlgebra
       const size_type begin = vector->getMap()->getMinGlobalIndex();
       const size_type end   = vector->getMap()->getMaxGlobalIndex() + 1;
 
-      Assert(
+      [[maybe_unused]] const size_type n_local_elements =
```

Comment:
I think normally we guard such things (when you do some work for an assert) with `#ifdef debug`.

Reply:
I guess the question is how much work is done in `getLocalNumElements`, or if it is a pure getter. In the latter case, the compiler will happily remove all code in release mode, but in case the definition is not visible, it will have to call the function. In consequence, the question boils down to what is less hassle for us in maintaining the code, to have an additional `#ifdef` or the present solution to suppress the warning.

Reply:
https://github.com/trilinos/Trilinos/blob/7001783146120820968a130cd0ad24ce7813bd88/packages/tpetra/core/src/Tpetra_Map_decl.hpp#L580-L587 shows that it's an inline getter.

Reply:
Here, we could of course also avoid the local variable altogether and just put the call to `get*NumElements()` right into the assertion...

Reply:
> Here, we could of course also avoid the local variable altogether and just put the call to `get*NumElements()` right into the assertion...

Only if we duplicate the whole assertion.

Reply:
Oh, that's because you can't do
```
  Assert (abc ==
#ifdef ...
               call this function()
#else
               call some other function()
#endif
             , Exc...());
```
?

Reply:
Yes, that's exactly what was there before. 🙂